borrowings/views.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+
+from django.http import HttpResponseRedirect
+from rest_framework import mixins, viewsets, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from borrowings.models import Borrowing
+from borrowings.serializers import (
+    BorrowingSerializer,
+    BorrowingListSerializer,
+    BorrowingDetailSerializer, BorrowingCreateSerializer,
+    BorrowingReturnSerializer
+)
+
+
+
+
+class BorrowingViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Borrowing.objects.select_related("book", "user")
+    serializer_class = BorrowingSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return BorrowingListSerializer
+
+        if self.action == "retrieve":
+            return BorrowingDetailSerializer
+
+        if self.action == "create":
+            return BorrowingCreateSerializer
+        if self.action == "return_book":
+            return BorrowingReturnSerializer
+
+        return BorrowingSerializer
+
```

Comment:
With which action is expected to return BorrowingSerializer? 
I think, you don't need it at all